Sources/BrowserServicesKit/Suggestions/SuggestionProcessing.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  Ordering.swift
+//
+//  Copyright © 2021 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Class encapsulates the whole ordering and filtering algorithm
+/// It takes query, history, bookmarks, and apiResult as input parameters
+/// The output is instance of SuggestionResult
+final class SuggestionProcessing {
```

Comment:
I expect there will be iterative changes of the ordering/filtering algorithm. My idea was to keep it ‘under one roof’ so it’s easy to change. We can split it when it’s more settled.